native-schema-registry/csharp/AWSGsrSerDe/AWSGsrSerDe/deserializer/DataFormatDeserializerFactory.cs

```diff
@@ -20,30 +38,32 @@ public static DataFormatDeserializerFactory GetInstance()
             return _dataFormatDeserializerFactoryInstance ??= new DataFormatDeserializerFactory();
         }
 
-        private DataFormatDeserializerFactory()
-        {
-        }
-
         /// <summary>
         /// Lazy initializes and returns a specific de-serializer instance.
         /// </summary>
         /// <param name="dataFormat">dataFormat for creating appropriate instance</param>
         /// <param name="configs">configuration elements for de-serializers</param>
         /// <returns>protocol specific de-serializer instance.</returns>
-        /// <exception cref="AwsSchemaRegistryException">Unsupported Data format</exception>
+        /// <exception cref="AwsSchemaRegistryException">Unsupported Data format.</exception>
         public IDataFormatDeserializer GetDeserializer(string dataFormat, GlueSchemaRegistryConfiguration configs)
         {
             return dataFormat switch
             {
-                nameof(GlueSchemaRegistryConstants.DataFormat.AVRO)=> GetAvroDeserializer(configs),
-                _ => throw new AwsSchemaRegistryException($"Unsupported data format: {dataFormat}")
+                nameof(GlueSchemaRegistryConstants.DataFormat.AVRO) => GetAvroDeserializer(configs),
+                nameof(GlueSchemaRegistryConstants.DataFormat.PROTOBUF) => GetProtobufDeserializer(configs),
+                _ => throw new AwsSchemaRegistryException($"Unsupported data format: {dataFormat}"),
             };
         }
 
-
         private AvroDeserializer GetAvroDeserializer(GlueSchemaRegistryConfiguration configs)
         {
             return _avroDeserializer ??= new AvroDeserializer(configs);
         }
+
+        private ProtobufDeserializer GetProtobufDeserializer(GlueSchemaRegistryConfiguration configs)
+        {
+            // TODO: store and reuse Deserializer for the same message
+            return new ProtobufDeserializer(configs.ProtobufMessageParser);
```

Comment:
Why don't we return same instance like Avro?

Reply:
Refactored